dwave/preprocessing/include/dwave/exceptions.hpp

```diff
@@ -0,0 +1,26 @@
+// Copyright 2023 D-Wave Systems Inc.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+
+#pragma once
+
+#include <stdexcept>
+#include <string>
+
+namespace dwave::presolve {
+
+struct InvalidModelError : public std::runtime_error {
```

Comment:
That will be what's conveyed in the `msg`. E.g. It will be used for things like
```c++
throw InvalidModelError("linear biases cannot contain NAN");
```

Reply:
this part ```This is disctinct from infeasibility.``` clarified my doubts, great!

Reply:
Yes. we still may want an `InfeasibleModelError` later... but we may want to instead set a flag. So for now I am leaving it out.